### PR TITLE
Private/caolan/async dns tidy

### PR DESCRIPTION
**Pull request by caolanm**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: net/AsyncDNS.hpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ class AsyncDNS
 
     void resolveDNS();
     void addLookup(const std::string& lookup,
-                   const std::optional<std::string>& port,
+                   const std::string& port,
                    const DNSThreadFn& cb,
                    const DNSThreadDumpStateFn& dumpState);
```
Comment (sgothel) [inline net/AsyncDNS.hpp:72]: mandatory port arg, good.

File: net/NetUtil.cpp [hunk 2]
```diff
@@ -129,23 +96,46 @@ HostEntry::HostEntry(const std::string& desc, const char* port)
...
-    initFromHostName(desc, port);
-}
+    addrinfo hints;
+    std::memset(&hints, 0, sizeof(hints));
+    hints.ai_flags = AI_CANONNAME | AI_ADDRCONFIG;
 
-HostEntry::~HostEntry()
-{
+    addrinfo* ainfo = nullptr;
+    int rc = getaddrinfo(desc.c_str(), port, &hints, &ainfo);
```
Comment (sgothel) [inline net/NetUtil.cpp:104]: ::getaddrinfo(..) ? (global namespace)

Comment (sgothel) [reply net/NetUtil.cpp:104]: personally I force myself to use ::system_call(..) to hint to me it is such a global function and not a method by any chance. Just taste.

File: net/AsyncDNS.hpp (hunk 1 above)
Comment (sgothel) [reply net/AsyncDNS.hpp:72]: OK, port string can be empty(), so still optional.

**Review by sgothel — APPROVED** [review]
looks good. removal of optional reduces the argument check as string.empty() check was missing. Plus added assertion message is helpful.

File: net/NetUtil.cpp (hunk 2 above)
Comment (caolanm) [reply net/NetUtil.cpp:104]: yeah, reasonable

**Pull request merged**